dascore/proc/whitening.py

```diff
@@ -0,0 +1,161 @@
+"""Signal whitening for time-domain applications."""
+from __future__ import annotations
+
+import math
+
+import numpy as np
+import numpy.fft as nft
+
+from dascore.constants import PatchType
+from dascore.exceptions import ParameterError
+from dascore.utils.patch import patch_function
+
+
+@patch_function(required_dims=("time", "distance"))
+def whitening(
+    patch: PatchType,
+    freq_range: [None, None] | float = None,
+    freq_smooth_size: None | float = None,
+) -> PatchType:
```

Comment:
To make this more general, we can do something similar to `pass_filter` which allows users to specify the dimension along which the filter is applied. Maybe the signature would look like this:

```python 
def whitening(
    patch: PatchType,
    smooth_size=None,
    **kwargs,
)
```

Then kwargs would be used to specify the dimension and frequency range (or None if 0 to nyquist). 

e.g., 

```python 
# whiten in freq. domain
patch.whitening(time=(10, 120))

# whiten in wave number domain
patch.whitening(distance=(1, 10))

# use 0 to nyquist for whitening
patch.whitening(time=None)
```

Reply:
Good idea, done.